spec.md

```diff
@@ -1386,6 +1540,8 @@ with a reset using the `regreset`{.firrtl} syntax.
 
 ### Registers without Reset
 
+TODO: Do I remove this?
```

Comment:
why would we remove this?

Reply:
In the next few weeks, I want to reorganize a bunch of things in the doc.

The chapter on "Statements" is not the proper grouping. It maybe made sense at one point. But this section is doing much more work than it should be. 

Rather than going statement-by-statement, I am going to move us towards explaining things concept-by-concept. The section I added on Circuit Components causes some tension, begging material from the Statements section to be moved up front. 

Until I have the re-organization finished, some material might get split up (with appropriate cross-references).